### ISSUE TITLE
What are all comments I can add to my code that turn on/off linters?

### ISSUE THREAD

**Issue by coreyog** [golangci/golangci-lint#754]
I'm not having any problems with the linter directly, but when I'm writing code that I don't want to be linted, I can't find documentation for what my options are for how to mark the code that way.

I've found that adding `// nolint` to the end of a line of code ensures that line isn't linted by any of the linters. I've seen in other submitted issues you can tell certain linters not to lint something with a comment like `// nolint:gocyclo`. It appears that adding either of these comments on the line before a function or `const(...)` block causes the whole block to be skipped.

What if I want to mark an arbitrary block of code to avoid linting? What if I want to disable multiple, but not all, linters? Can I mark an entire file to be skipped during linting? Are there any other features I don't even know to ask about?

Basically, what comment structures does the linter look for and how do they behave (e.g. putting `// nolint` on a line with nothing else before a function definition indicates the entire function shouldn't be linted, that makes sense but isn't obvious that I should expect that)?

Having a document discussing this added to the project's wiki would be great.

**Comment by jirfag** [comment]
https://github.com/golangci/golangci-lint/blob/master/README.md#nolint
